### PR TITLE
Upgrade devfile api to newest with deprecated v1alpha1

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Upgrade devfile api to newest with deprecated v1alpha1.
It currently depends on commit from https://github.com/devfile/api/pull/524, but after it's merged revision should be updated to one from main branch.

### What issues does this PR fix or reference?
depends on https://github.com/devfile/api/pull/524

It's for https://github.com/devfile/devworkspace-operator/issues/471

### Is it tested? How?
```
export DWO_IMG=my-custom
make install
kc apply -f ./samples/theia-next.yaml
# check that you're able to get v1alpha1 as well as v1alpha2, but v1alpha1 will lead to a warning:
```
Screenshot_20210706_165606

Also I checked that Che with devworkspace support works as well.

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v7-devworkspaces-operator-e2e, v7-devworkspace-happy-path` to trigger)
    - [ ] `v7-devworkspaces-operator-e2e`: DevWorkspace e2e test
    - [ ] `v7-devworkspace-happy-path`: DevWorkspace e2e test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -190,19 +191,21 @@ endif
...
+	go get -d -v sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
+	go build -o $(GOBIN)/controller-gen-$(CONTROLLER_GEN_VERSION) sigs.k8s.io/controller-tools/cmd/controller-gen
 	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
 	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
 else
-CONTROLLER_GEN=$(shell which controller-gen)
+	@echo "Using installed $(GOBIN)/controller-gen-$(CONTROLLER_GEN_VERSION)"
 endif
+CONTROLLER_GEN=$(GOBIN)/controller-gen-$(CONTROLLER_GEN_VERSION)
```
Comment (amisevsk) [inline Makefile:208]: If this is outside of a `if/else/endif`, we should move to the top of the file. This line is processed before any rules anyways, so having it here just creates confusion IMO.

File: deploy/deployment/kubernetes/objects/devworkspaceroutings.controller.devfile.io.CustomResourceDefinition.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.3.0
+    controller-gen.kubebuilder.io/version: v0.6.1
```
Comment (amisevsk) [inline deploy/deployment/kubernetes/objects/devworkspaceroutings.controller.devfile.io.CustomResourceDefinition.yaml:5]: I think some generated files are missing in this PR. When I run the generate commands locally, these changes also get propagated to `deploy/bundle/manifests/controller.devfile.io_devworkspaceroutings.yaml`.

Another +1 for https://github.com/devfile/devworkspace-operator/pull/449 :)

**Review by amisevsk — COMMENTED** [review]
Mostly LGTM, but I haven't had a chance to test the full operator install flow.

File: deploy/deployment/kubernetes/objects/devworkspaceroutings.controller.devfile.io.CustomResourceDefinition.yaml (hunk 2 above)
Comment (sleshchenko) [reply deploy/deployment/kubernetes/objects/devworkspaceroutings.controller.devfile.io.CustomResourceDefinition.yaml:5]: if you meant missing defaults, now it's fixed. Please recheck on your end.

File: Makefile (hunk 1 above)
Comment (sleshchenko) [reply Makefile]: fixed.

**Review by amisevsk — APPROVED** [review]
Tested locally on `crc` and looks good -- well done :)

**Comment by sleshchenko** [comment]
The referenced PR is merged into main branch, so it's updated

**Review by JPinkney — APPROVED** [review]
Just tested on crc and everything looks good to me

**Comment by sleshchenko** [comment]
Happy path is failing due blocker on Che side, which is being solved.
Merging without it as I believe it should not break Che integration.

**Pull request merged**